libelement/include/element/interpreter.h

```diff
@@ -252,6 +252,18 @@ element_result element_instruction_is_constant(
     element_instruction* instruction,
     bool* constant);
 
+/**
+ * @brief converts an instruction in to a human-readable string
+ *
+ * @param[in] instruction       instruction to represent as a string
+ * @param[out] buffer           output buffer
+ * @param[in,out] buffer_size   output buffer size, modified to represent the number of characters written. if buffer is NULL, the required buffer size will be written
+ */
+element_result element_instruction_to_string(
+    element_instruction* instruction,
+    char* buffer,
+    int* buffer_size);
```

Comment:
Any reason we're using `int` here rather than `size_t`? Just guarding against potential underflow?

Reply:
I think my habit of using signed integers comes from things like `ssize` being a thing in future C++ versions, and maybe some conference talks I've watched? can't really remember :b generally I feel like the extra ~48bits isn't used, so ints are always better as you mentioned, due to underflow, but it definitely makes API's non-intuitive and currently results in lots of added static_casts from size_t's.

I've changed it in the latest commit + the other stuff we talked about, should be good now :)

decided not to allocate/deallocate the buffer internally despite resulting in double the work, as the performance of the function isn't that important for debugging, and having char* in C that a user can't delete is likely prone to error.